src/Interfaces/hamiltonians.jl

```diff
@@ -419,3 +449,47 @@ See also [`LOStructure`](@ref Main.Hamiltonians.LOStructure).
 has_adjoint(op) = has_adjoint(LOStructure(op))
 has_adjoint(::AdjointUnknown) = false
 has_adjoint(::LOStructure) = true
+
+"""
+    OperatorColumn(op::AbstractOperator{T}, address::A) <: AbstractVector{Pair{A,T}}
+
+Default column, using `offdiagonals(op, address)` and `diagonal_element(op, address)`.
+"""
+struct OperatorColumn{A,T,O<:Union{AbstractOperator{T},AbstractMatrix{T}},OD} <: AbstractVector{Pair{A,T}}
+    operator::O
+    address::A
+    ods::OD
+    diagonal::T
+end
+
+"""
+    operator_column(op::AbstractOperator, address)
+
+Return an object representing the column of `op` given by `address`. By default, returns
+[`OperatorColumn(op, address, offdiagonals(op, address), diagonal_element(op, address))`](@ref).
```

Comment:
```suggestion
Return an object representing the column of `op` given by `address`. 
```
I suggest to not document the default implementation, as that is an implementation detail that we may want to change later (without generating a new major version).